saleor/graphql/product/utils.py

```diff
@@ -33,42 +35,63 @@ def validate_attributes_input_for_product_and_page(
         code=error_code_enum.REQUIRED.value,
     )
     error_dropdown_get_more_than_one_value = ValidationError(
-        "Attribute attribute must take only one value",
-        code=error_code_enum.INVALID.value,
+        "Attribute must take only one value", code=error_code_enum.INVALID.value,
     )
     error_blank_value = ValidationError(
         "Attribute values cannot be blank", code=error_code_enum.REQUIRED.value,
     )
 
+    error_no_file_given = ValidationError(
+        "Attribute file url cannot be blank", code=error_code_enum.REQUIRED.value,
+    )
+    error_blank_file_value = ValidationError(
+        "Attribute expects a file url but none were given",
+        code=error_code_enum.REQUIRED.value,
+    )
+
     attribute_errors: Dict[ValidationError, List[str]] = defaultdict(list)
-    for attribute, values in input_data:
-        attribute_id = graphene.Node.to_global_id("Attribute", attribute.pk)
-        if not values:
-            if attribute.value_required:
-                attribute_errors[error_no_value_given].append(attribute_id)
-            continue
-
-        if attribute.input_type != AttributeInputType.MULTISELECT and len(values) != 1:
-            attribute_errors[error_dropdown_get_more_than_one_value].append(
-                attribute_id
-            )
-            continue
-
-        for value in values:
+    for attribute, attr_values in input_data:
+        attribute_id = attr_values.global_id
+        # validation for file attribute
+        if attribute.input_type == AttributeInputType.FILE:
```

Comment:
Maybe not for now, but when we will have more input types it would be nice to extract validation specific to a type to functions, then we would only have something like:
```
if attribute.input_type == AttributeInpuType.FOO:
    validate_foo_input(...)
elif attribute.input_type == AttributeInpuType.BAR:
    validate_bar_input(...)
```

Reply:
Yes, I was thinking about that. When the next type appears it will be necessary.